src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx

```diff
@@ -163,134 +166,138 @@ export const ContentMessageComponent = ({
-            key={asset.type}
-            asset={asset}
-            message={message}
-            selfId={selfId}
-            onClickButton={onClickButton}
-            onClickImage={onClickImage}
-            onClickMessage={onClickMessage}
+        {failedToSend && (
+          <PartialFailureToSendWarning
             isMessageFocused={msgFocusState}
-            is1to1Conversation={conversation.is1to1()}
-            isLastDeliveredMessage={isLastDeliveredMessage}
-            onClickDetails={() => onClickDetails(message)}
-          />
-        ))}
-
-        {isAssetMessage && (
-          <ReadIndicator
-            message={message}
-            is1to1Conversation={conversation.is1to1()}
-            isLastDeliveredMessage={isLastDeliveredMessage}
-            onClick={onClickDetails}
+            failedToSend={failedToSend}
+            knownUsers={conversation.allUserEntities()}
           />
         )}
 
-        {!isConversationReadonly && isActionMenuVisible && (
-          <MessageActionsMenu
-            isMsgWithHeader={!hideHeader}
-            message={message}
-            handleActionMenuVisibility={setActionMenuVisibility}
-            contextMenu={contextMenu}
-            isMessageFocused={msgFocusState}
+        {!!reactions.length && (
+          <MessageReactionsList
+            reactions={reactions}
+            selfUserId={selfId}
             handleReactionClick={onClickReaction}
-            reactionsTotalCount={reactions.length}
+            isMessageFocused={msgFocusState}
+            onTooltipReactionCountClick={() => onClickReactionDetails(message)}
+            onLastReactionKeyEvent={() => setActionMenuVisibility(false)}
             isRemovedFromConversation={conversation.removed_from_conversation()}
+            users={conversation.allUserEntities()}
           />
         )}
       </div>
-
-      {[StatusType.FAILED, StatusType.FEDERATION_ERROR].includes(status) && (
-        <CompleteFailureToSendWarning
-          {...(status === StatusType.FEDERATION_ERROR && {unreachableDomain: conversation.domain})}
-          isMessageFocused={msgFocusState}
-          onRetry={() => onRetry(message)}
-        />
-      )}
-
-      {failedToSend && (
-        <PartialFailureToSendWarning
-          isMessageFocused={msgFocusState}
-          failedToSend={failedToSend}
-          knownUsers={conversation.allUserEntities()}
-        />
-      )}
-
-      {!!reactions.length && (
-        <MessageReactionsList
-          reactions={reactions}
-          selfUserId={selfId}
-          handleReactionClick={onClickReaction}
-          isMessageFocused={msgFocusState}
-          onTooltipReactionCountClick={() => onClickReactionDetails(message)}
-          onLastReactionKeyEvent={() => setActionMenuVisibility(false)}
-          isRemovedFromConversation={conversation.removed_from_conversation()}
-          users={conversation.allUserEntities()}
-        />
-      )}
-    </div>
+      <DeliveredIndicator
```

Comment:
Another option was to add it on the level of the `MessageWrapper`, but I'm not sure it won't break again the stretching of the message for content like `pre`, and maybe we will need completely different behaviour for different kinds of messages. Also, it avoids additional `div` containers